Handle expired session based on rememberMe token

tiagomgelias opened the pull request.

tiagomgelias committed:
Handle expired session based on rememberMe token

claudio-silva commented on subsystems/authentication/Authentication/Middleware/AuthenticationMiddleware.php:
```diff
@@ -27,34 +31,56 @@ class AuthenticationMiddleware implements RequestHandlerInterface
   function __invoke (ServerRequestInterface $request, ResponseInterface $response, callable $next)
   {
     $this->redirection->setRequest ($request);
+    $serverRequest = ServerRequest::fromGlobals ();
```
Isto não pode ser nem faz sentido. Estás a passar por cima de todo o ciclo PSR-7, mais valia usares $_COOKIE 👎 . Usa o argumento $request.

claudio-silva commented on subsystems/authentication/Authentication/Middleware/AuthenticationMiddleware.php:
```diff
@@ -27,34 +31,56 @@ class AuthenticationMiddleware implements RequestHandlerInterface
+
       $this->session->logout ();
-      return $this->redirection->home ();
+      if ($cookies->has ($this->kernelSettings->name . "/" . $this->kernelSettings->rememberMeTokenName)) {
```
1. Usa $sessionSettings->cookieName
2. A não ser que haja uma razão especial para usares `/`, prefiro que uses `_`, já que `/` é um delimitador de URL.
3. Mete o nome do cookie numa variável para não repetires várias vezes o cálculo do nome (princípio DRY = Don't Repeat Yourself)

claudio-silva commented on subsystems/authentication/Authentication/Middleware/AuthenticationMiddleware.php:
```diff
@@ -27,34 +31,56 @@ class AuthenticationMiddleware implements RequestHandlerInterface
-
-        if (!$this->session->loggedIn ())
+        if (!$this->session->loggedIn ()) {
+          if ($cookies->has ($this->kernelSettings->name . "/" . $this->kernelSettings->rememberMeTokenName)) {
```
Então e se o cookie de remember não existir?
Pode não existir por ter sido apagado do browser, ou por a funcionalidade de remember estar desactivada (quer por checkbox no login, quer por ter sido desactivada na config da app).
Tens de verificar estas possibilidades todas.
Não criar cookie se tal não fizer sentido.
Mas se fizer, onde crias o cookie?

tiagomgelias committed:
Change of settings variable
-Changes in getting all cookies, instead of request being global, the request used is that comes through controller

tiagomgelias committed:
Cookie name at 1 variable (DRY = Don't Repeat Yourself)

tiagomgelias committed:
Cookie name variable in 1 line
